api/client/webclient/webclient_test.go

```diff
@@ -489,13 +490,18 @@ func TestWebClientClosesIdleConnections(t *testing.T) {
 	require.NoError(t, err)
 	require.Empty(t, cmp.Diff(expectedResponse, resp))
 
-	for _, expected := range expectedStates {
+	var got []string
+	for i := range expectedStates {
 		select {
 		case state := <-stateChange:
-			require.Equal(t, expected, state, "expected connection state %s got %s", expected.String(), state.String())
+			got = append(got, state)
 		case <-time.After(3 * time.Second):
-			t.Fatalf("timeout waiting for expected connection state %s", expected.String())
+			t.Fatalf("timeout waiting for expected connection state %d", i)
 		}
 	}
 
+	slices.Sort(expectedStates)
+	slices.Sort(got)
+
+	require.Equal(t, expectedStates, got)
```

Comment:
can't you use `require.ElementsMatch(t, []int{1, 2, 3, 4}, []int{4, 3, 2, 5})`? 
It ignores the order of the events and ensures the two lists include the same values

Reply:
It's kinda the same but preserves the ordering which might be important when debugging why a certain test is wrong.